tests/test_plugins.py

```diff
@@ -17,14 +17,14 @@ def setUp(self):
             template="page.html",
             language=self.language,
         )
-        self.home.publish(self.language)
         self.page = create_page(
             title="content",
             template="page.html",
             language=self.language,
         )
-        self.page.publish(self.language)
-        self.placeholder = self.page.placeholders.get(slot="content")
+        self.placeholder = self.page.get_placeholders(
+            self.language
```

Comment:
Here you would need two different cases: for django CMS < 4, it is just `get_placeholders()` due to different model structures. Something like:
```
if cms_version < "4":
    self.placeholder = self.page.get_placeholders(self.language).get(slot="content")
else:
    self.placeholder = self.page.get_placeholders().get(slot="content")
```

and import `from cms import __version__ as cms_version` at the top.